packages/nextjs/src/common/wrapGenerationFunctionWithSentry.ts

```diff
@@ -33,88 +32,82 @@ export function wrapGenerationFunctionWithSentry<F extends (...args: any[]) => a
   const { requestAsyncStorage, componentRoute, componentType, generationFunctionIdentifier } = context;
   return new Proxy(generationFunction, {
     apply: (originalFunction, thisArg, args) => {
-      const requestTraceId = getActiveSpan()?.spanContext().traceId;
-      return escapeNextjsTracing(() => {
-        let headers: WebFetchHeaders | undefined = undefined;
-        // We try-catch here just in case anything goes wrong with the async storage here goes wrong since it is Next.js internal API
-        try {
-          headers = requestAsyncStorage?.getStore()?.headers;
-        } catch (e) {
-          /** empty */
-        }
+      let headers: WebFetchHeaders | undefined = undefined;
+      // We try-catch here just in case anything goes wrong with the async storage here goes wrong since it is Next.js internal API
+      try {
+        headers = requestAsyncStorage?.getStore()?.headers;
+      } catch (e) {
+        /** empty */
+      }
 
-        let data: Record<string, unknown> | undefined = undefined;
-        if (getClient()?.getOptions().sendDefaultPii) {
-          const props: unknown = args[0];
-          const params = props && typeof props === 'object' && 'params' in props ? props.params : undefined;
-          const searchParams =
-            props && typeof props === 'object' && 'searchParams' in props ? props.searchParams : undefined;
-          data = { params, searchParams };
-        }
+      const isolationScope = commonObjectToIsolationScope(headers);
 
-        const headersDict = headers ? winterCGHeadersToDict(headers) : undefined;
+      const activeSpan = getActiveSpan();
+      if (activeSpan) {
+        const rootSpan = getRootSpan(activeSpan);
+        rootSpan.setAttribute('sentry.rsc', true);
+        setCapturedScopesOnSpan(rootSpan, new Scope(), isolationScope);
```

Comment:
m: Should we not clone the current `currentScope` here, instead of doing `new Scope()`? 🤔

Reply:
Updated to grab the current scope off the root span, so we do not stomp it.